test/integration/connection-monitoring-and-pooling/connection.test.js

```diff
@@ -452,6 +452,26 @@ describe('Connection', function () {
       })
     );
 
+    it('throws when attempting an operation if the client is not connected', function (done) {
+      const client = this.configuration.newClient();
+      const collection = client.db('shouldCorrectlyFailOnRetry').collection('test');
+      collection.insertOne({ a: 2 }, err => {
+        expect(err).to.match(/must be connected/);
+        done();
+      });
+    });
+
+    it('throws when attempting an operation if the client is not connected (promises)', async function () {
+      const client = this.configuration.newClient();
+      const collection = client.db('shouldCorrectlyFailOnRetry').collection('test');
+
+      try {
+        await collection.insertOne({ a: 2 });
```

Comment:
We need an expect.fail() below the insertOne to make sure the error doesn't disappear on us.

An alternative:
```ts
const error = await collection.insertOne({a:2}).catch(error => error)
expect(error).to.be.instanceOf(MongoNotConnectedError)
```
You can keep it all a oneliner and it catches both cases here, if it succeeds the instanceOf will fail, if it throws we get our error returned to us.